perf/impl/https/v0.1/main.go

```diff
@@ -0,0 +1,199 @@
+package main
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/binary"
+	"encoding/json"
+	"encoding/pem"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"math/big"
+	"net"
+	"net/http"
+	"time"
+)
+
+const (
+	BlockSize = 64 << 10
+)
+
+func handleRequest(w http.ResponseWriter, r *http.Request) {
+	// Read the big-endian bytesToSend value
+	var bytesToSend uint64
+	if err := binary.Read(r.Body, binary.BigEndian, &bytesToSend); err != nil {
+		http.Error(w, "failed to read uint64 value", http.StatusBadRequest)
+		return
+	}
+
+	// Read and discard the remaining bytes in the request
+	io.Copy(io.Discard, r.Body)
+
+	buf := make([]byte, BlockSize)
+
+	for bytesToSend > 0 {
+		toSend := buf
+		if bytesToSend < BlockSize {
+			toSend = buf[:bytesToSend]
+		}
+
+		n, err := w.Write(toSend)
+		if err != nil {
+			http.Error(w, "Failed write", http.StatusInternalServerError)
+			return
+		}
+		bytesToSend -= uint64(n)
+	}
+}
+
+func runClient(serverAddr string, uploadBytes, downloadBytes uint64) (time.Duration, time.Duration, error) {
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		},
+	}
+
+	reqBody := make([]byte, 8+uploadBytes)
+	binary.BigEndian.PutUint64(reqBody, uint64(downloadBytes))
+
+	startTime := time.Now()
+	resp, err := client.Post(fmt.Sprintf("https://%s/", serverAddr), "application/octet-stream", bytes.NewReader(reqBody))
+	if err != nil {
+		return 0, 0, err
+	}
+	uploadDoneTime := time.Now()
```

Comment:
Do you know if this is actually the time when the bytes have been flushed?

Reply:
I think it's worth just measuring the whole thing from start to finish rather than trying to measure the parts. It'll be more accurate since we know for sure when we're done versus when we've finished the upload.

Reply:
That applies to the non-HTTP version as well. The transport may buffer the data before flushing it (quic-go does, albeit just for a single MTU worth of data, so it won't really matter).